packages/office-ui-fabric-react/src/components/Nav/Nav.tsx

```diff
@@ -216,9 +216,10 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
   private _isLinkSelected(link: INavLink): boolean {
     // if caller passes in selectedKey, use it as first choice or
     // if current state.selectedKey (from addressbar) is match to the link
-    if (this.props.selectedKey !== undefined && link.key === this.props.selectedKey ||
-        this.state.selectedKey !== undefined && link.key === this.state.selectedKey) {
-      return true;
+    if (this.props.selectedKey !== undefined ) {
+      return link.key === this.props.selectedKey;
+    } else if (this.state.selectedKey !== undefined) {
+      return link.key === this.state.selectedKey;
```

Comment:
We discussed this:

The remaining logic in this code won't execute after this change once a single link has been clicked.

So the question is, should we even keep the remaining logic? I feel like we should remove it as it should really be controlled by the caller. Here's how I would expect the Nav links to work:

1. Each INavLink should take in a `selected` prop. If this is not undefined, we should use this as the truth.
2. The Nav itself should take `selectedKey`. If this is provided, we will fall back to match the link key with this key.
3. If the caller wants the selected link to change based on URL changes, we should perhaps let the caller handle that and keep the logic out of the component.